File: src/lecture12/java/Lecture12Main.java
package lecture12.java;

import lecture12.kotlin.Person;

public class Lecture12Main {

    public static void main(String[] args) {
        //Person.Companion.newBaby("ABC");
        Person.newBaby("ABC");
        Person.Factory.newBaby("ABC222");

        moveSomting(new Movable() {

            @Override
            public void move() {
                System.out.println("움직인다.");
            }

            @Override
            public void fly() {
                System.out.println("난다.");
            }
        });
    }
    private static void moveSomting(Movable movable) {
        movable.move();
        movable.fly();
    }
}

File: src/lecture12/kotlin/Lecture12Main.kt
package lecture12.kotlin

import lecture12.java.Movable

fun main() {
    moveSomething(object : Movable {
        override fun move() {
            println("움직")
        }

        override fun fly() {
            println("날날")
        }

    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}